game/scripts/vscripts/internal/events.lua

```diff
@@ -84,6 +84,8 @@ function GameMode:_OnEntityKilled( keys )
 
     if not killerPlayerEntity:IsRealHero() then
       killerPlayerEntity = killerEntity:GetOwnerEntity() or killerEntity
+      --Change respawn timer for hero killed by creep.
+      killedUnit:SetTimeUntilRespawn(5)   
```

Comment:
i think you should create a new conditional and check the team of the killer. this would also resolve the last of the non-hero-entities problems if in the else block you put the hero killed event.